src/app/application.cpp

```diff
@@ -279,12 +280,34 @@ void Application::torrentFinished(BitTorrent::TorrentHandle *const torrent)
         program.replace("%T", torrent->currentTracker());
         program.replace("%I", torrent->hash());
 
+        logger->addMessage(tr("Torrent: %1, running external program: %2").arg(torrent->name()).arg(program));
+#if defined(Q_OS_UNIX)
+        QProcess::startDetached(QLatin1String("/bin/sh"), {QLatin1String("-c"), program});
+#elif defined(Q_OS_WIN32)
+        // test cmd: `echo "%F" > "c:\ab ba.txt"`
+        STARTUPINFOW si = {0};
+        si.cb = sizeof(si);
+        PROCESS_INFORMATION pi = {0};
+
+        program = program.prepend(QLatin1String("cmd.exe /C ")).left(MAX_PATH - 1);
+        WCHAR *arg = new WCHAR[program.size() + 1];
+        program.toWCharArray(arg);
```

Comment:
>Isn't it enough to call `QString::toUtf16()` instead?

Apparently, you meant  `QString::utf16()`... But this is not entirely correct. It returns `ushort` array.